pkg/controller/core/queuedworkload_controller.go

```diff
@@ -200,3 +238,66 @@ func workloadFinished(w *kueue.QueuedWorkload) bool {
 	}
 	return false
 }
+
+// GetWorkloadCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
+func GetWorkloadCondition(status *kueue.QueuedWorkloadStatus, conditionType kueue.QueuedWorkloadConditionType) (int, *kueue.QueuedWorkloadCondition) {
+	if status == nil {
+		return -1, nil
+	}
+	return GetWorkloadConditionFromList(status.Conditions, conditionType)
+}
+
+// GetWorkloadConditionFromList extracts the provided condition from the given list of condition and
+// returns the index of the condition and the condition. Returns -1 and nil if the condition is not present.
+func GetWorkloadConditionFromList(conditions []kueue.QueuedWorkloadCondition, conditionType kueue.QueuedWorkloadConditionType) (int, *kueue.QueuedWorkloadCondition) {
+	if conditions == nil {
+		return -1, nil
+	}
+	for i := range conditions {
+		if conditions[i].Type == conditionType {
+			return i, &conditions[i]
+		}
+	}
+	return -1, nil
+}
+
+// UpdateWorkloadStatus is a wrapper to update the condition of a workload
+func (r *QueuedWorkloadReconciler) UpdateWorkloadStatus(ctx context.Context,
+	wl *kueue.QueuedWorkload,
+	wlct kueue.QueuedWorkloadConditionType, conditionStatus corev1.ConditionStatus,
+	message, reason string) error {
+	conditionIndex, oldCondition := GetWorkloadCondition(&wl.Status, wlct)
+
+	now := metav1.Now()
+	condition := kueue.QueuedWorkloadCondition{
+		Type:               wlct,
+		Status:             conditionStatus,
+		LastProbeTime:      now,
+		LastTransitionTime: now,
+		Reason:             reason,
+		Message:            message,
+	}
+	var newWl kueue.QueuedWorkload
+
+	if oldCondition == nil {
+		// We are adding new pod condition.
+		wl.Status.Conditions = append(wl.Status.Conditions, condition)
+		return r.client.Status().Update(ctx, wl)
+	}
+
+	// We are updating an existing condition, so we need to check if it has changed.
+	if conditionIndex != -1 {
+		if err := r.client.Get(ctx, client.ObjectKeyFromObject(wl), &newWl); err != nil {
```

Comment:
Just use the object `wl` that you already have.

Reply:
that's why the comment on line 289, `We are updating an existing condition, so we need to check if it has changed` 
removing the `if` on 290 it would look like this
```go
	// We are updating an existing condition, so we need to check if it has changed.
	if err := r.client.Get(ctx, client.ObjectKeyFromObject(wl), &newWl); err != nil {
		return err
	}
	if newWl.Status.Conditions[conditionIndex].Status == wl.Status.Conditions[conditionIndex].Status &&
		newWl.Status.Conditions[conditionIndex].Type == wl.Status.Conditions[conditionIndex].Type {
		// No need to update
		return nil
	}
	newWl.Status.Conditions[conditionIndex] = condition
```